pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -117,12 +117,18 @@ func (o *SelectorOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 	o.mapper = mapper
 	o.encoder = f.JSONEncoder()
 
-	o.builder = f.NewBuilder().
+	o.builder = f.NewBuilder(!o.local).
 		ContinueOnError().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, &o.fileOptions).
 		Flatten()
 
+	if !o.local {
```

Comment:
I am not sure what you mean by this. 
The reason why `o.builder.Latest()` is in this block is because invoking this method causes visitors to make use of the [RetrieveLatest](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/builder.go#L783) visitor function, which then attempts to [make use of the `RESTClient` for each info](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/visitor.go#L117). Since this `RESTClient` is `nil` when running the command `local`ly ([due to the resource `ClientMapper` being decorated](https://github.com/kubernetes/kubernetes/pull/45918/files#diff-a9b5082d3df09126fbcd118712148907R157) with [DisabledClientForMapping](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/mapper.go#L35)), not having this block results in a panic.